pkgs/misc/vim-plugins/vim-utils.nix

```
@@ -485,7 +486,18 @@ rec {
     '';
   };
 
-  inherit (import ./build-vim-plugin.nix { inherit lib stdenv rtpPath vim; }) buildVimPlugin buildVimPluginFrom2Nix;
+  vimGenDocHook = callPackage ({ vim }:
+    makeSetupHook {
+      name = "vim-gen-doc-hook";
+      deps = [ vim ];
+      substitutions = {
+        vimBinary = "${vim}/bin/vim";
+        inherit rtpPath;
+      };
+    } ./vim-gen-doc-hook.sh) {};
+
+  inherit (import ./build-vim-plugin.nix { inherit lib stdenv rtpPath vim vimGenDocHook; })
+    buildVimPlugin buildVimPluginFrom2Nix;
```

since you brought in callPackage (I'm assuming from the vimPlugins scope), you should be able to do 
```suggestion
  inherit (callPackage ./build-vim-plugin.nix { inherit rtpPath vim vimGenDocHook; })
    buildVimPlugin buildVimPluginFrom2Nix;
```

but it's not like it's strictly nicer, so I'm largely indifferent

---

thanks for the review. As you mention, it doesn't look strictly nicer so I will pass to avoid having to recheck everything